lib/jobs.ts

```diff
@@ -0,0 +1,217 @@
+               dct:modified ${sparqlEscapeDateTime(now)} .
+    }
+  }`);
+
+  return {
+    id: jobUuid,
+    uri: jobUri,
+    status: config.job.statuses.scheduled,
+    created: now,
+    modified: now
+  };
+}
+
+async function getReportIds(job) {
+  const result = await query(`
+  PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
+  PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
+  PREFIX dct: <http://purl.org/dc/terms/>
+  PREFIX prov: <http://www.w3.org/ns/prov#>
+  PREFIX adms: <http://www.w3.org/ns/adms#>
+
+  SELECT ?report ?reportId
+  WHERE {
+    GRAPH <${config.job.graph}> {
+      ${sparqlEscapeUri(job.uri)} prov:used ?report .
+      ?report mu:uuid ?reportId .
+    }
+  }`);
+  const bindings = result.results.bindings;
+  if (bindings.length > 0) {
+    return bindings.map((binding) => binding['reportId'].value);
+  } else {
+    return [];
+  }
+}
+
+async function getNextScheduledJob() {
+  const result = await query(`
+  PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
+  PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
+  PREFIX dct: <http://purl.org/dc/terms/>
+  PREFIX adms: <http://www.w3.org/ns/adms#>
+
+  SELECT ?uri ?id ?status
+  WHERE {
+    GRAPH <${config.job.graph}> {
+      VALUES ?status {
+        ${sparqlEscapeUri(config.job.statuses.scheduled)}
+      }
+      ?uri a ext:ReportGenerationJob ;
+           mu:uuid ?id ;
+           dct:created ?created ;
+           adms:status ?status .
+      FILTER NOT EXISTS {
+        ?job a ext:ReportGenerationJob ;
+           adms:status ${sparqlEscapeUri(config.job.statuses.ongoing)} .
+      }
+    }
+  } ORDER BY ASC(?created) LIMIT 1`);
+
+  const bindings = result.results.bindings;
+  if (bindings.length === 1) {
+    let job = {
+      id: bindings[0]['id'].value,
+      uri: bindings[0]['uri'].value,
+      status: bindings[0]['status'].value,
+    };
+    job.reportIds = await getReportIds(job);
+    return job;
+  } else {
+    return null;
+  }
+}
+
+export async function getJob(jobId) {
+  const result = await query(`
+  PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
+  PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
+  PREFIX dct: <http://purl.org/dc/terms/>
+  PREFIX adms: <http://www.w3.org/ns/adms#>
+
+  SELECT ?uri ?id ?status ?created ?modified
+  WHERE {
+    GRAPH <${config.job.graph}> {
+      ?uri a ext:ReportGenerationJob ;
+           mu:uuid ${sparqlEscapeString(jobId)} ;
+           dct:created ?created ;
+           dct:modified ?modified ;
+           adms:status ?status .
+    }
+  } ORDER BY ASC(?created) LIMIT 1`);
+
+  const bindings = result.results.bindings;
+  if (bindings.length === 1) {
+    let job = {
+      id: jobId,
+      uri: bindings[0]['uri']?.value,
+      status: bindings[0]['status']?.value,
+      created: bindings[0]['created']?.value,
+      modified: bindings[0]['modified']?.value,
+    };
+    job.reportIds = await getReportIds(job);
+    return job;
```

Comment:
**issue (non-blocking)**: This return value is used as a response body, but it's not JSON:API compliant

The implicit contract of a semantic-works service is that it speaks JSON:API, to maintain a standard and to leverage common tools (e.g. Ember Data can load in the JSON:API payload as a record, like we do here https://github.com/kanselarij-vlaanderen/frontend-kaleidos/blob/development/app/utils/zip-agenda-files.js#L41-L47)

Given the time sensitivity of these issues I'm prone to saying we just let it go through as-is and revisit it later (granted, later might end up being never)

Reply:
> later might end up being never

Absolutely. Will fix now

Reply:
Guessing we didn't do this? Later it is. rest is good to go